src/GitObjectDb/Models/Rebase/ObjectRepositoryRebase.cs

```diff
@@ -21,49 +17,43 @@ namespace GitObjectDb.Models.Rebase
 {
     /// <inheritdoc />
     [ExcludeFromGuardForNull]
-    public class ObjectRepositoryRebase : IObjectRepositoryRebase
+    internal class ObjectRepositoryRebase : IObjectRepositoryRebase
     {
-        private readonly IServiceProvider _serviceProvider;
         private readonly IRepositoryProvider _repositoryProvider;
         private readonly IObjectRepositoryLoader _repositoryLoader;
         private readonly MigrationScaffolderFactory _migrationScaffolderFactory;
+        private readonly RebaseProcessor.Factory _rebaseProcessorFactory;
 
         private readonly RepositoryDescription _repositoryDescription;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ObjectRepositoryRebase"/> class.
         /// </summary>
-        /// <param name="serviceProvider">The service provider.</param>
         /// <param name="container">The container.</param>
         /// <param name="repositoryDescription">The repository description.</param>
         /// <param name="repository">The repository on which to apply the merge.</param>
         /// <param name="rebaseCommitId">The commit to be rebased.</param>
         /// <param name="branchName">Name of the branch.</param>
-        /// <exception cref="ArgumentNullException">
-        /// serviceProvider
-        /// or
-        /// repositoryDescription
-        /// or
-        /// commitId
-        /// or
-        /// branchName
-        /// or
-        /// merger
-        /// </exception>
+        /// <param name="repositoryProvider">The repositoyr provider.</param>
+        /// <param name="repositoryLoader">The repository loader.</param>
+        /// <param name="migrationScaffolderFactory">The <see cref="MigrationScaffolder"/> factory.</param>
+        /// <param name="rebaseProcessorFactory">The <see cref="MergeProcessor"/> factory.</param>
         [ActivatorUtilitiesConstructor]
-        public ObjectRepositoryRebase(IServiceProvider serviceProvider, IObjectRepositoryContainer container, RepositoryDescription repositoryDescription, IObjectRepository repository, ObjectId rebaseCommitId, string branchName)
+        public ObjectRepositoryRebase(IObjectRepositoryContainer container, RepositoryDescription repositoryDescription, IObjectRepository repository, ObjectId rebaseCommitId, string branchName,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Constructor has 9 parameters, which is greater than the 7 authorized. ([csharpsquid:S107](https://sonarcloud.io/organizations/frblondin-github/rules#rule_key=csharpsquid%3AS107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=GitObjectDb&pullRequest=33&issues=AWe4KJcjD1JUB5M3-M2i&open=AWe4KJcjD1JUB5M3-M2i)

[ISSUE_KEY]: AWe4KJcjD1JUB5M3-M2i